src/components/steps/StepBillingData.tsx

```diff
@@ -268,8 +273,14 @@ export default function StepBillingData({
                 <Grid item xs={12}>
                   <Typography>
                     <Checkbox
-                      checked={formik.values.publicServices ? true : false}
+                      checked={stepHistoryState.isGSP}
```

Comment:
The checked value should be ruled by the publicServices field, not the isGSP condition